libnopegl/src/text.h

```diff
@@ -89,14 +98,34 @@ struct text_cls {
     uint32_t flags; // combination of NGLI_TEXT_FLAG_*
 };
 
+/* Each field points to a contiguous data buffer (1 row per character) */
+struct text_effects_pointers {
+    float *transform; // mat4[]
+    float *color;     // vec3[]
+    float *opacity;   // f32[]
+};
+
+struct effect_segmentation {
+    size_t *positions;     // character index (in chars darray) to position in "target unit" (char, word, ...)
+    size_t total_segments; // total number of segment: all values in positions are between [0;total_segments-1]
+};
+
 struct text {
     struct ngl_ctx *ctx;
     struct text_config config;
 
+    /* user exposed */
     int32_t width;
     int32_t height;
     struct darray chars; // struct char_info
     struct texture *atlas_texture;
+    struct text_effects_pointers data_ptrs; // set of effect data pointers (in chars_data)
+
+    /* effects specific */
```

Comment:
private ?

Reply:
Mmh I prefer specifying that the following block refers to effects specific fields; there are other private fields that do not relate to effects all around

Reply:
Ok.